ui/packages/catalog-ui-search/src/main/webapp/component/map-info/map-info.view.js

```diff
@@ -117,24 +117,23 @@ module.exports = Marionette.LayoutView.extend({
     )
   },
   getAttributes() {
-    if (this.model.get('targetMetacard') === undefined) {
-      return []
-    }
     const jsx = []
-    properties.summaryShow.forEach(attribute => {
-      const attributeName = hbHelper.getAlias(attribute)
-      const attributeValue = this.model
-        .get('targetMetacard')
-        .get('metacard')
-        .get('properties')
-        .get(attributeName)
-      if (attributeValue !== undefined)
-        jsx.push(
-          <div className="info-feature">
-            {formatAttribute(attributeName, attributeValue)}
-          </div>
-        )
-    })
+    if (this.model.get('targetMetacard') !== undefined) {
+      properties.summaryShow.forEach(attribute => {
+        const attributeName = hbHelper.getAlias(attribute)
+        const attributeValue = this.model
+          .get('targetMetacard')
+          .get('metacard')
+          .get('properties')
+          .get(attributeName)
+        if (attributeValue !== undefined)
+          jsx.push(
```

Comment:
This array pushing, then returning the array syntax is quite different from the rest of our react code. Have you tried just returning `props.summaryShow.map(attribute => <...>` where in `...` you are checking if `attributeValue` is undefined, and if so you just return null?

Reply:
That's pre-existing code. I simply restructured the method to return `[]` by default instead of conditionally exiting the function early. Additionally, using arrays seems cleaner than returning nulls. In a few cases I've seen nulls in jsx trigger warnings in React.